mdata/cache/chunk_cache.go

```diff
@@ -0,0 +1,96 @@
+package cache
+
+import (
+	"flag"
+	"sync"
+
+	"github.com/raintank/metrictank/iter"
+	"github.com/rakyll/globalconf"
+)
+
+var (
+	maxSize uint64
+)
+
+func ConfigSetup() {
+	flags := flag.NewFlagSet("chunk-cache", flag.ExitOnError)
+	// (1024 ^ 3) * 4 = 4294967296
+	flags.Uint64Var(&maxSize, "max-size", 4294967296, "Maximum size of chunk cache in bytes")
+	globalconf.Register("chunk-cache", flags)
+}
+
+type CacheChunk struct {
+	Ts      uint32
+	Next    uint32
+	Prev    uint32
+	itergen iter.IterGen
+}
+
+type MetricCache struct {
```

Comment:
So how about this:

* `CCache` for the whole cache struct
* `CCacheMetric` for each of the metrics inside the `CCache`
* `CCacheChunk` for each of the chunks inside a `CCacheMetric`

That seems quite consistent, not too long, easy to type, easy to understand. Do you think so?